srcpkgs/devilutionX/template

```diff
@@ -1,16 +1,25 @@
 # Template file for 'devilutionX'
 pkgname=devilutionX
-version=1.2.1
+version=1.4.1
 revision=1
 build_style=cmake
-configure_args="-DVERSION_NUM=$version -DBINARY_RELEASE=ON -DTTF_FONT_PATH=\"/usr/share/fonts/truetype/CharisSILB.ttf\""
-makedepends="SDL2-devel SDL2_ttf-devel SDL2_mixer-devel libsodium-devel"
+configure_args="-DVERSION_NUM=$version -DBINARY_RELEASE=ON -DBUILD_TESTING=OFF -DDISABLE_ZERO_TIER=ON"
+hostmakedepends="git gettext pkg-config smpq"
+makedepends="SDL2-devel SDL2_image-devel fmt-devel libpng-devel libsodium-devel bzip2-devel"
 short_desc="Diablo I engine for modern operating systems"
 maintainer="MarcoAPC <marcoaureliopc@gmail.com>"
 license="Unlicense"
 homepage="https://github.com/diasurgical/devilutionX"
-distfiles="https://github.com/diasurgical/devilutionX/archive/${version}.tar.gz"
-checksum=002dcbd4d4a5bdf8db1a3ec01139e5bfbed46d6a1caa32b17c9f2df161ad3521
+distfiles="https://github.com/diasurgical/devilutionX/archive/${version}.tar.gz
+https://raw.githubusercontent.com/diasurgical/devilutionX/1.2.1/CMake/FindSDL2.cmake"
```

Comment:
It builds perfectly without `FindSDL2.cmake`. Remove.

Reply:
You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated, it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories, due to the problem with find SDL2 on various cpus architectures.

https://pkgs.org/download/devilutionx-32bit
https://pkgs.org/download/devilutionx

Reply:
> You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated

No, it does not make maintenance complicated. In fact, this is practiced here.

In fact, if you really want to download git revisions by abusing bandwidth every single time on every clean rebuild - be my guest. I am not sure whether committers will let this go off the hook easily.

> it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories

How come if it is not easy to compile after 1.2.1 when I could, Marco?

> due to the problem with find SDL2 on various cpus architectures.

Do you have any proof that SDL2 is not found "on various cpu architectures"?

How can you explain SDL2 being found here on void (be it native or cross) successfully without this cursed file in my case?

The urls provided by you are not convincing, i.e. not telling anything at all.

Reply:
> You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated, it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories, due to the problem with find SDL2 on various cpus architectures.

If CMake is fetching submodules at build time, that is unacceptable. Void templates should explicitly fetch all components of a build so that we can verify that the contents are as expected and the process is as reproducible as possible.

Reply:
> > You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated
> 
> No, it does not make maintenance complicated. In fact, this is practiced here.
> 
Yes, it's complicated, it's a simple game, not a gcc template version.
> If you really want to download git revisions by abusing bandwidth every single time on every clean rebuild - be my guest. I am not sure whether committers will let this go off the hook easily.
> 
I applied the suggestions from the Void devs, each suggestion came from a different dev, so I had to apply and resubmit the PR. Besides, version 1.2.1 always had static packages, this is nothing new.
> > it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories
> 
> How come it is not easy to compile after 1.2.1 when I could, Marco?
> 
From the 1.3.x series, the FindSDL2.cmake file is removed, so it only compiles for i686 and x86_64, the 1.2.x series had arm binary package for Linux in the github of devilutionX, in the 1.3.x series it was removed.

Then try compiling the 1.3.x series for non-x86 architectures, using the 1.2.1 template as a reference!
> > due to the problem with find SDL2 on various cpus architectures.
> 
> Do you have any proof that SDL2 is not found "on various cpu architectures"?
> 
Try compiling the 1.3.x series for non-x86, using the default cmake configuration from deviltutionx.
> How can you explain SDL2 being found here on void (be it native or cross) successfully without this cursed file in my case?
> 
You have disabled the option to download static software and download them manually, so there is no SDL2 check.
> The urls provided by you are not convincing, i.e. not telling anything at all.

You didn't look at the links, there are only three versions of devilutionX 1.2.1; 1.4.0 and 1.4.1.

  Why does version 1.2.1 exist in distro repositories?

Reply:
> If CMake is fetching submodules at build time, that is unacceptable. Void templates should explicitly fetch all components of a build so that we can verify that the contents are as expected and the process is as reproducible as possible.

It's not a submodule! The same process exists since version 1.2.1.

Reply:
That means we've been doing it wrong and it needs to be fixed.

Reply:
> That means we've been doing it wrong and it needs to be fixed.

So in software made in GO and rust, is it necessary to create a template that downloads all the libraries manually, creating unnecessary lines in the template, just because the libraries don't come with source code?

What can be done is to add software that is in the Void repositories in "hostmakedepends" and "makedepends" or maybe create new packages that are not in the repositories, this is the only way to prevent cmake from downloading static software libraries, they are not submodules , I consider this approach unnecessary, as the template is ready.

Reply:
Consider it unnecessary if you want, but that's what we expect.